File: cal.c
#include <stdio.h>
#include <stdlib.h>

double evaluate_expression(const char* expression);

int main() {
    char expression[100];

    printf("Enter an expression: ");
    scanf("%s", expression);

    double result = evaluate_expression(expression);
    printf("Result: %lf\n", result);

    return 0;
}

double evaluate_expression(const char* expression) {
    char operators[100];
    double values[100];
    int op_top = -1, val_top = -1;
    const char* ptr = expression;

    while (*ptr) {
        if (*ptr >= '0' && *ptr <= '9') {
            double num = 0;
            while (*ptr >= '0' && *ptr <= '9') {
                num = num * 10 + (*ptr - '0');
                ptr++;
            }
            if (*ptr == '.') {
                ptr++;
                double frac = 0.1;
                while (*ptr >= '0' && *ptr <= '9') {
                    num += (*ptr - '0') * frac;
                    frac /= 10;
                    ptr++;
                }
            }
            values[++val_top] = num;
        } else if (*ptr == '+' || *ptr == '-' || *ptr == '*' || *ptr == '/') {
            while (op_top != -1 && 
                  ((operators[op_top] == '*' || operators[op_top] == '/') ||
                   ((operators[op_top] == '+' || operators[op_top] == '-') && (*ptr == '+' || *ptr == '-')))) {
                double b = values[val_top--];
                double a = values[val_top--];
                char op = operators[op_top--];
                switch (op) {
                    case '+': values[++val_top] = a + b; break;
                    case '-': values[++val_top] = a - b; break;
                    case '*': values[++val_top] = a * b; break;
                    case '/': values[++val_top] = a / b; break;
                }
            }
            operators[++op_top] = *ptr;
            ptr++;
        } else {
            ptr++;
        }
    }

    while (op_top != -1) {
        double b = values[val_top--];
        double a = values[val_top--];
        char op = operators[op_top--];
        switch (op) {
            case '+': values[++val_top] = a + b; break;
            case '-': values[++val_top] = a - b; break;
            case '*': values[++val_top] = a * b; break;
            case '/': values[++val_top] = a / b; break;
        }
    }

    return values[val_top];
}

File: cal.cs
using System;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        Console.Write("Enter an expression: ");
        string expression = Console.ReadLine();

        double result = EvaluateExpression(expression);
        Console.WriteLine("Result: " + result);
    }

    static double EvaluateExpression(string expression)
    {
        Stack<double> values = new Stack<double>();
        Stack<char> operators = new Stack<char>();
        int i = 0;

        while (i < expression.Length)
        {
            if (char.IsDigit(expression[i]))
            {
                double num = 0;
                while (i < expression.Length && char.IsDigit(expression[i]))
                {
                    num = num * 10 + (expression[i] - '0');
                    i++;
                }
                if (i < expression.Length && expression[i] == '.')
                {
                    i++;
                    double frac = 0.1;
                    while (i < expression.Length && char.IsDigit(expression[i]))
                    {
                        num += (expression[i] - '0') * frac;
                        frac /= 10;
                        i++;
                    }
                }
                values.Push(num);
            }
            else if (expression[i] == '+' || expression[i] == '-' || expression[i] == '*' || expression[i] == '/')
            {
                while (operators.Count > 0 &&
                      ((operators.Peek() == '*' || operators.Peek() == '/') ||
                       ((operators.Peek() == '+' || operators.Peek() == '-') && (expression[i] == '+' || expression[i] == '-'))))
                {
                    double b = values.Pop();
                    double a = values.Pop();
                    char op = operators.Pop();
                    switch (op)
                    {
                        case '+': values.Push(a + b); break;
                        case '-': values.Push(a - b); break;
                        case '*': values.Push(a * b); break;
                        case '/': values.Push(a / b); break;
                    }
                }
                operators.Push(expression[i]);
                i++;
            }
            else
            {
                i++;
            }
        }

        while (operators.Count > 0)
        {
            double b = values.Pop();
            double a = values.Pop();
            char op = operators.Pop();
            switch (op)
            {
                case '+': values.Push(a + b); break;
                case '-': values.Push(a - b); break;
                case '*': values.Push(a * b); break;
                case '/': values.Push(a / b); break;
            }
        }

        return values.Peek();
    }
}